spec/requests/adminpanel_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe '/adminpanel', type: :request do
+  before(:each) do
+    user = User.create!(email: 'test@example.com', username: 'test', password: 'f4k3p455w0rd', user_type: 'administrator')
+    user.confirm
+    login_as(user, scope: :user)
+  end
+
+  describe 'GET /index' do
+    it 'renders a successful response' do
+      users = User.create(email: 'example@example.com', username: 'example', password: 'dhsjhgd67', user_type: 'administrator')
+      get adminpanel_url(users)
+      expect(response).to be_successful
+    end
+  end
+
+  describe 'GET /edit' do
+    it 'render a successful response' do
+      users = User.create!(email: 'example@example.com', username: 'example', password: 'dhsjhgd67', user_type: 'administrator')
+      get edit_adminpanel_path(users)
+      expect(response).to be_successful
+    end
+  end
+
+  describe 'PATCH /update' do
+    context 'with valid parameters' do
+      let(:new_attributes) {
+        { user_id: '7', username: 'example1', user_type: 'user' }
+      }
+
+      it 'updates the requested user' do
+        user = User.create!(email: 'example2@example.com', username: 'example', password: 'dhsjhgd67', user_type: 'administrator')
```

Comment:
user is repeated in https://github.com/BC-MAR-21-ROR/proyecto-denuncia-ciudadana-proyecto-team03/pull/19/files#diff-70483a97c768368008169d358acb470cb8e05f1c5fcb7a9d98e68303b5fdd6ffR39
could be better have `user` defined outside and reuse in both specs